crates/websocket/src/send.rs

```diff
@@ -0,0 +1,162 @@
+use crate::{
+  messages::{SendComment, SendCommunityRoomMessage, SendPost, SendUserRoomMessage},
+  LemmyContext,
+  OperationType,
+};
+use lemmy_api_common::{
+  blocking,
+  comment::CommentResponse,
+  community::CommunityResponse,
+  person::PrivateMessageResponse,
+  post::PostResponse,
+};
+use lemmy_db_queries::DeleteableOrRemoveable;
+use lemmy_db_schema::{CommentId, CommunityId, LocalUserId, PersonId, PostId, PrivateMessageId};
+use lemmy_db_views::{
+  comment_view::CommentView,
+  local_user_view::LocalUserView,
+  post_view::PostView,
+  private_message_view::PrivateMessageView,
+};
+use lemmy_db_views_actor::community_view::CommunityView;
+use lemmy_utils::{ConnectionId, LemmyError};
+
+pub async fn send_post_ws_message<OP: ToString + Send + OperationType + 'static>(
```

Comment:
These are all very scary to me, because there could have been slight differeces that got lost in the ones you moved here. And since errors wouldn't show up in the unit tests, since the tests use the HTTP API, we can't guarantee these are all good. 

So please test all the websocket actions before you merge.

Reply:
Anyway I made a lot of changes which we need to test, so I'd prefer to test it all at once before we release.

Reply:
mmmk